packages/apollo-server-caching/src/__tests__/connectors.test.ts

```diff
@@ -0,0 +1,32 @@
+import { servers, delay } from '../__mocks__/common';
+import MemcachedKeyValueCache from '../connectors/memcached';
+import RedisKeyValueCache from '../connectors/redis';
+import each from 'jest-each';
+
+// run test suite against each implementation of KeyValueCache
+each([
+  [new MemcachedKeyValueCache(servers.memcachedHost)],
+  [new RedisKeyValueCache({ host: servers.redisHost })],
+]).describe('%s', keyValueCache => {
+  beforeEach(() => {
+    keyValueCache.flush();
+  });
+
+  afterAll(() => {
+    keyValueCache.close();
+  });
+
+  it('can do a basic get and set', async () => {
+    await keyValueCache.set('hello', 'world');
+    expect(await keyValueCache.get('hello')).toBe('world');
+    expect(await keyValueCache.get('missing')).not.toBeDefined();
+  });
+
+  it('is able to expire keys based on ttl', async () => {
+    await keyValueCache.set('short', 's', { ttl: 1 });
+    await keyValueCache.set('long', 'l', { ttl: 5 });
+    await delay(1500);
```

Comment:
Having to depend on actual time passing (as opposed to mocking timers) is another reason this isn't really suitable as a unit test.

Reply:
Is it acceptable to leave out tests for key expiry?

Reply:
This may blast the complexity of the tests, but I can recommend lolex for doing time stuff. You can see an example here: https://github.com/n1ru4l/react-time-ago/blob/29006c7a2080626ec4206560fe4ecf69c3ca079b/src/index.test.js#L52

Reply:
For sure! I will test it out with the mocked databases and it will be great it that plays well together! Thanks 👍

Reply:
Great! Yeah, as long as we're using mocks for the databases we don't need actual delays.